test/integration/controlplane/transformation/transformation_test.go

```diff
@@ -254,43 +256,27 @@ func (e *transformTest) getRawSecretFromETCD() ([]byte, error) {
 func (e *transformTest) getEncryptionOptions(reload bool) []string {
 	if e.transformerConfig != "" {
 		return []string{
-			"--encryption-provider-config", path.Join(e.configDir, encryptionConfigFileName),
+			"--encryption-provider-config", filepath.Join(e.configDir, encryptionConfigFileName),
 			fmt.Sprintf("--encryption-provider-config-automatic-reload=%v", reload),
 			"--disable-admission-plugins", "ServiceAccount"}
 	}
 
 	return nil
 }
 
-func (e *transformTest) createEncryptionConfig(ecSymLink bool) (string, string, error) {
-	tempDir, err := os.MkdirTemp("", "secrets-encryption-test")
-	if err != nil {
-		return "", "", fmt.Errorf("failed to create temp directory: %v", err)
-	}
-
-	if ecSymLink {
-		// create another temp dir
-		parentTempDir, err := os.MkdirTemp("", "secrets-encryption-symlink-test")
-		if err != nil {
-			return tempDir, "", fmt.Errorf("failed to create temp directory: %v", err)
-		}
-
-		// create config file
-		if err := os.WriteFile(filepath.Join(parentTempDir, encryptionConfigFileName), []byte(e.transformerConfig), 0644); err != nil {
-			return tempDir, parentTempDir, fmt.Errorf("failed to write encryption config file: %v", err)
-		}
-
-		// create symlink
-		if err := os.Symlink(filepath.Join(parentTempDir, encryptionConfigFileName), filepath.Join(tempDir, encryptionConfigFileName)); err != nil {
-			return tempDir, parentTempDir, fmt.Errorf("failed to create symlink: %v", err)
-		}
+func (e *transformTest) createEncryptionConfig() (
+	filePathForEncryptionConfig,
+	symLinkPathForEncryptionConfig string,
+	err error,
+) {
+	tempDir := filepath.Join(os.TempDir(), "symlinkdir")
```

Comment:
Any reason for not using `os.MkdirTemp`?

Reply:
updated